runners/flink/src/main/java/org/apache/beam/runners/flink/PipelineTranslationRewriter.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.flink;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import org.apache.beam.runners.core.SplittableParDoViaKeyedWorkItems;
+import org.apache.beam.runners.core.construction.PTransformMatchers;
+import org.apache.beam.runners.core.construction.PTransformReplacements;
+import org.apache.beam.runners.core.construction.SingleInputOutputOverrideFactory;
+import org.apache.beam.runners.core.construction.SplittableParDo;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.runners.AppliedPTransform;
+import org.apache.beam.sdk.runners.PTransformOverride;
+import org.apache.beam.sdk.transforms.Combine;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.View;
+import org.apache.beam.sdk.util.InstanceBuilder;
+import org.apache.beam.sdk.values.PCollection;
+
+/**
+ * A {@link Pipeline.PipelineVisitor} for overriding user {@link PTransform PTransforms}
+ * with {@link FlinkRunner}-specific implementations.
+ */
+public class PipelineTranslationRewriter extends FlinkPipelineTranslator {
+
+  private final FlinkRunner flinkRunner;
+  private final boolean isStreaming;
+
+  public PipelineTranslationRewriter(FlinkRunner flinkRunner) {
+    this.flinkRunner = checkNotNull(flinkRunner, "flinkRunner");
+    this.isStreaming = flinkRunner.getPipelineOptions().isStreaming();
+  }
+
+  @Override
+  public void translate(Pipeline pipeline) {
+    pipeline.replaceAll(getOverrides());
+  }
+
+  private List<PTransformOverride> getOverrides() {
+    if (isStreaming) {
```

Comment:
Can this not just be inlined in the appropriate spot? We can have a utility class for the overrides if we want, but I don't think it's hugely useful, and we can call the pipeline.replaceAll(overrides) right before trying to translate via the streaming and/or batch pipeline translator. I don't think this additional abstraction layer gets us anything.